lib/humidifier/aws_adapters/base.rb

```diff
@@ -31,6 +31,12 @@ def update(payload)
         try_valid { client.update_stack(payload.update_params) }
       end
 
+      # Upload a CFN stack to S3 so that it can be referenced via template_url
+      def upload(payload)
+        Humidifier.config.ensure_upload_configured!(payload)
+        upload_object(payload, "#{Humidifier.config.s3_prefix}#{payload.identifier}.json")
```

Comment:
Will `s3_prefix` always have a trailing slash? Or will `payload.identifier` always have a leading slash? If not, I think we should account for that case to make sure this path is valid